docs/src/main/asciidoc/datasource-guide.adoc

```diff
@@ -0,0 +1,212 @@
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-mariadb</artifactId>
+</dependency>
+--
+[source,xml]
+--
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+--
+
+== Configuring the datasource
+
+After all of the dependencies are added to your pom.xml file, you'll need to configure Agroal.
+This is done in `{config-file}`.
+The minimum necessary for a configured database is as follows:
+
+[source,properties]
+--
+include::../../../../extensions/agroal/deployment/src/test/resources/application-named-datasource.properties[]
+--
+
+There are other configuration options, detailed below.
+
+== Injecting a Datasource
+
+Because {project-name} uses CDI, injecting a Datasource is very simple:
+
+[source,java,indent=0]
+--
+include::../../../../extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/DefaultDataSourceConfigTest.java[tag=injection]
+--
+
+In the above example, the type is `AgroalDataSource` which is a subtype of `javax.sql.DataSource`.
+Because of this, you can also use `javax.sql.DataSource`.
+
+== Multiple Datasources
+
+Agroal allows you configure multiple datasources.
+It works exactly the same way as a single datasource, with one important change: a name.
+
+[source,properties]
+--
+include::../../../../extensions/agroal/deployment/src/test/resources/application-multiple-datasources.properties[]
+--
+
+Notice there's an extra bit in the key.
+The syntax is as follows: `quarkus.datasource.[optional name.][datasource property]`.
+
+=== Named Datasource Injection
+
+When using multiple datasources, each `DataSource` also has the `Named` qualifier with the name of the datasource in the property as the value.
+Using the above properties to configure three different datasources, you can also inject each one as follows:
+
+[source,java,indent=0]
+--
+include::../../../../extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/MultipleDataSourcesConfigTest.java[tag=injection]
+--
+
+== Agroal Configuration Reference
+
+|===
+|Configuration key|Java type|Example
+
+|quarkus.datasource.driver
+|String (class name)
+|org.h2.Driver, com.mysql.jdbc.Driver
+
+|quarkus.datasource.url
+|String (JDBC Connection URL)
+|jdbc:h2:tcp://localhost/mem:default, jdbc:mysql://hostname:port/dbname
+
+|quarkus.datasource.username
+|String
+|Fred, Bill, inventory_user
+
+|quarkus.datasource.password
+|String
+|correct horse battery staple, Tr0ub4dor&3
+
+
+|quarkus.datasource.min-size
+|Integer
+|5, 12, 42
+
+|quarkus.datasource.max-size
+|Integer
+|5, 12, 42
+
+|quarkus.datasource.initial-size
+|Integer
+|5, 12, 42
+
+|quarkus.datasource.background-validation-interval
+|java.time.Duration
+|PT3M, PT56S
+
+|quarkus.datasource.acquisition-timeout
+|java.time.Duration
+|PT3M, PT56S
+
+|quarkus.datasource.leak-detection-interval
+|java.time.Duration
+|PT3M, PT56S
+
+|quarkus.datasource.idle-removal-interval
+|java.time.Duration
+|PT3M, PT56S
```

Comment:
I would maybe add a note below about this format as it might not be entirely obvious.